src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -98,7 +98,11 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
 
             if (!document.Project.Solution.Workspace.Options.GetOption(CompletionOptions.BlockForCompletionItems, service.Language))
             {
-                _textView.Options.GlobalOptions.SetOptionValue(NonBlockingCompletionEditorOption, true);
+                // The Editor supports the option per textView.
+                // There could be mixed desired behavior per textView and even per same completion session.
+                // The right fix would be to send this information as a result of the method. 
+                // Then, the Editor would choose the right behavior for mixed cases.
+                _textView.Options.SetOptionValue(NonBlockingCompletionEditorOption, true);
```

Comment:
also set to false so that, for example pure C#, gets blocking completion even though there was typescript somewhere in the file.

Reply:
Yes, this is correct. Thank you! Otherwise, the option will be read from GlobalSettings